pages/metadata/index.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <v-row class="pa-4">
+    <v-col cols="3" class="tree-view-container">
+      <v-treeview
+        :items="items"
+        :load-children="fetchItems"
+        activatable
+        transition
+        @update:active="handleAction"
+      ></v-treeview>
+    </v-col>
+
+    <v-col v-if="itemId" cols="9">
+      <metadata-editor :item-id="itemId"></metadata-editor>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+import { BaseItemDto } from '~/api';
+
+type ITreeNode = {
+  id: string | number | undefined;
+  name: string | null | undefined;
+  children?: ITreeNode[];
+};
+export default Vue.extend({
+  data() {
+    return {
+      items: [] as ITreeNode[],
+      itemId: ''
+    };
+  },
+  computed: {},
+  async created() {
+    const floders = (await this.$libraryApi.getMediaFolders()).data
+      .Items as BaseItemDto[];
+
+    this.items = floders.map((dir, index) => {
+      return {
+        id: dir.Id || index,
+        name: dir.Name,
+        children: [] as ITreeNode[]
+      };
+    });
+  },
+  methods: {
+    async fetchItems(node: ITreeNode) {
+      const userId = this.$auth.user.Id;
+      const libItems = (((
+        await this.$userLibraryApi.getItem(
+          { userId, itemId: '' },
+          {
+            query: {
+              ParentId: node.id,
+              SortBy: 'SortName'
+            }
+          }
+        )
+      ).data as unknown) as { Items: BaseItemDto[] }).Items; //
```

Comment:
This line seems really dirty.

Fixing the `$userLibraryApi` to the new format looks to be fixing the issues you have with types on my end.

Reply:
As a general rule of thumb, it's best to avoid forcing types on stuff from the API.

The entire purpose of using TypeScript and a generated API client is to warn us of API changes as soon as they hit.

Reply:
This interface does not support generics. The format returned by the interface depends on whether the itemId is empty, so there are two solutions:
1. Assert to the correct type, as I do now
2. Modify the type of getItem so that it supports generics.

But if the type file is automatically generated, no matter what kind of scheme, we can't prompt us when the interface changes.